app/src/main/java/teamagam/ConnectionTest/ConnectionTestFragment.java

```diff
@@ -60,17 +61,18 @@ private String getUrl() {
     }
 
     private void httpCheck(String url) {
-        ValidatorAsyncTask httpTask = new ValidatorAsyncTask(
-                new HttpValidator(),
-                new UIDisplayer(mHttpResultTextView));
-        httpTask.execute(url);
+        executeValidatorTask(url, new HttpValidator(), mHttpResultTextView);
     }
 
     private void pingCheck(String url) {
-        ValidatorAsyncTask pingTask = new ValidatorAsyncTask(
-                new PingValidator(),
-                new UIDisplayer(mPingResultTextView));
-        pingTask.execute(url);
+        executeValidatorTask(url, new PingValidator(), mPingResultTextView);
+    }
+
+    private void executeValidatorTask(
+            String url, ValidatorAsyncTask.Validator validator, TextView resultTextView) {
+        ValidatorAsyncTask task = new ValidatorAsyncTask(
+                validator, new UIDisplayer(resultTextView));
+        task.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, url);
```

Comment:
is that also solve the threads problem we spoke of yesterday?, if yes how? (I think this expansion is better saved face to face).